Upstream-Status: Backport

Part 1/3 of making the GN binary link correctly. More information:
https://crbug.com/934793

Signed-off-by: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
---
From 0ac93dd23bafdae516d4d5318c7b748c5c98b489 Mon Sep 17 00:00:00 2001
From: Khouri Giordano <kirmeo@gmail.com>
Date: Fri, 22 Feb 2019 11:01:35 -0500
Subject: [PATCH] For proper order put -lpthreads as a 'library' not a 'linker
 flag'

Linker flags appear early among the link command line arguments,
before other libraries that need pthreads. Changing it to a library
puts it at the end of the command line.

Change-Id: I2d0cc5e44a33d72f9fcaa67f3a6fbc0da565a735
Reviewed-on: https://gn-review.googlesource.com/c/4060
Reviewed-by: Brett Wilson <brettw@chromium.org>
Commit-Queue: Brett Wilson <brettw@chromium.org>
---
 build/gen.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/gen.py b/build/gen.py
index c426c0bf..c330abca 100755
--- a/tools/gn/build/gen.py
+++ b/tools/gn/build/gen.py
@@ -334,7 +334,7 @@ def WriteGNNinja(path, platform, host, options):
       ldflags.append('-maix64')
 
     if platform.is_posix():
-      ldflags.append('-lpthread')
+      libs.append('-lpthread')
 
     if options.use_lto:
       cflags.extend(['-flto', '-fwhole-program-vtables'])
-- 
2.20.1

